### DISCUSSION TITLE
Should RunJS have an optional sidebar?

### DISCUSSION THREAD

**Discussion by lukehaas** [lukehaas/RunJS#555] [Ideas]
![Screenshot 2023-03-30 at 16 27 01](https://user-images.githubusercontent.com/741344/228886653-7d50ac36-08fd-45b3-8daf-c38b186bb373.png)

A sidebar would provide a more suitable location for the `run` button than the current location. It would also make sense for a `stop` button to be provided.

**Comment by ghost** [comment]
For this purpose, I don't see the need! It would be interesting if it were to improve the organization of files, in directories. Runjs encompasses a larger layer than just a playground, but also a snippet manager.

Runjs has a very attractive and minimalist interface! However, the arrival of a sidebar would bring important gains, such as organizing files into directories, instead of having thousands of tabs open. Consequently improving performance.

I think of something like:

![Captura de Tela 2023-12-31 às 10 43 44](https://github.com/lukehaas/RunJS/assets/41877339/02080a50-d50f-4d06-8b98-4ae89eb37c78)

![Captura de Tela 2023-12-31 às 10 43 26](https://github.com/lukehaas/RunJS/assets/41877339/56a5f244-a775-4255-bfa8-e36d958fe9bf)

**Comment by palgramming** [comment]
wow this message was asked over a year ago and honestly the RUN button is the whole reason I came here looking for answers today. And I was going to suggest the the Run button be moved to the right of the Hamburger Menu.... But your side bar idea is nice maybe it would be a place that could allow snippets to be stored... Like highlight a group of text and drop it to the side bar as temp storage as one refactors their code not sure what other ideas you had envisioned to include in the side bar 

I think my RUN button issue is how my word wrap might be configured and as I typed my code going under the Run button but then if I adjusted window width then the code reformatted and was not longer hiding under the button

**Comment by lukehaas** [comment]
Thanks @palgramming. Glad to hear you like the idea. It's going to be included in the next version of RunJS.

**Reply by palgramming** [reply]
Great I will keep renewing my copies cause it is really a great tool and I am thankful for your time and effort as you keep constantly improving it

**Reply by lukehaas** [reply]
Thanks! If you have any more ideas or suggestions, please feel free to share them.
